pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -146,15 +146,18 @@ func countIntolerableTaintsPreferNoSchedule(taints []v1.Taint, tolerations []v1.
 
 // Score invoked at the Score extension point.
 func (pl *TaintToleration) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	s, err := getPreScoreState(state)
+	if err != nil {
+		return 0, framework.AsStatus(err)
+	}
+
 	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
 	if err != nil {
 		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w", nodeName, err))
 	}
 	node := nodeInfo.Node()
-
-	s, err := getPreScoreState(state)
-	if err != nil {
-		return 0, framework.AsStatus(err)
+	if node == nil {
```

Comment:
This is not necessary, if node is nil, we'll return error above. See
https://github.com/kubernetes/kubernetes/blob/8594df1aff9efde5f7e152b331093527eb9cf46a/pkg/scheduler/internal/cache/snapshot.go#L189-L194

Reply:
Thanks for your suggestions. I changed it.